src/compiler/compile/nodes/shared/Expression.ts

```diff
@@ -201,7 +201,7 @@ export default class Expression {
 					scope = map.get(node);
 				}
 
-				if (is_reference(node, parent)) {
+				if (node.type === 'Identifier' && is_reference(node, parent)) {
```

Comment:
not sure why in `is-reference` https://github.com/Rich-Harris/is-reference/blob/ca593f330bff5660efec747ddc5973e875e2f923/module.js#L2-L4, `MemberExpression` is treated as reference for certain scenario